### PR TITLE
fix 786 researchselect message dlgs rises twice

### LINKED ISSUE #786: [interface] "Project too large", etc... message dialogues rises twice
When I try to start construction of large item in small workshop the error message dialogue shows twice.

It's non critical interface bug.

### PR DESCRIPTION
**Pull request by Atrosha**
fix #786

### CODE REVIEW

**Comment by Atrosha** [comment]
It seems `AlwaysEmitSelectionEvents` in

https://github.com/OpenApoc/OpenApoc/blob/ff1db57d65abdb1ee0d4c5c2051801da254104bb/forms/listbox.cpp#L184-L195

isn't properly implemented.

**Comment by Atrosha** [comment]
And now this dialogues doesn't shows up after clicking again before changing selection, but this is much better then was previously.

**Comment by FilmBoy84** [comment]
Tested and appears to work, though with the caveat Atrosha mentions in #870 
I'm unsure whether to merge with the rest of the code still being a mess though?

File: game/ui/base/researchselect.cpp [hunk 1]
```diff
@@ -80,31 +79,32 @@ void ResearchSelect::begin()
 			if (topic->isComplete())
 			{
 				LogInfo("Topic already complete");
-				auto message_box = mksp<MessageBox>(tr("PROJECT COMPLETE"),
-				                                    tr("This project is already complete."),
-				                                    MessageBox::ButtonOptions::Ok);
-				fw().stageQueueCommand({StageCmd::Command::PUSH, message_box});
+				fw().stageQueueCommand({StageCmd::Command::PUSH,
```
Comment (JonnyH) [inline game/ui/base/researchselect.cpp:82]: To my eye these blocks look functionally identical?

Am I missing a change?

If just a cleanup I don't know if it's necessary, or at least should be a separate commit to functional changes.

**Comment by JonnyH** [comment]
Yeah, this all looks a bit spidery - I really hate magic undocumented bool flags changing behavior of things....

Might be worth digging down and seeing how much of this complexity is necessary?

**Comment by Atrosha** [comment]
I just remove `AlwaysEmitSelectionEvents = true` and container variables for dialogues.

File: game/ui/base/researchselect.cpp (hunk 1 above)
Comment (Atrosha) [reply game/ui/base/researchselect.cpp:82]: I just remove AlwaysEmitSelectionEvents = true and container variables for dialogues.

**Comment by Atrosha** [comment]
@emptyVoid found better solution for this. Closing.
